start

```diff
@@ -472,20 +461,58 @@ function init_soroban_rpc() {
-    fi
-    if [ ".$PROTOCOL_VERSION" != ".none" ] ; then
-      if [ $PROTOCOL_VERSION -gt 0 ]; then
-        curl "http://localhost:11626/upgrades?mode=set&upgradetime=1970-01-01T00:00:00Z&protocolversion=$PROTOCOL_VERSION&basereserve=5000000"
-      fi
-    fi
+  if [ "$NETWORK" != "local" ]; then
+    return
+  fi
+
+  # Wait for server
+  while ! echo "Stellar-core http server listening!" | nc localhost 11626 &> /dev/null; do sleep 1; done
+
+  # Default to latest version supported by core
+  if [ -z "$PROTOCOL_VERSION" ]; then
+    export PROTOCOL_VERSION=`curl -s http://localhost:11626/info | jq -r '.info.protocol_version'`
+  fi
+
+  if [ ".$PROTOCOL_VERSION" == ".none" ] ; then
+    return
   fi
+
+  # Upgrade local network's protocol version and base reserve to match pubnet/testnet
+  if [ $PROTOCOL_VERSION -gt 0 ]; then
+    echo "upgrades: protocolversion=$PROTOCOL_VERSION, basereserve=5000000"
+    curl -s "http://localhost:11626/upgrades?mode=set&upgradetime=1970-01-01T00:00:00Z&protocolversion=$PROTOCOL_VERSION&basereserve=5000000"
+    while [ "`curl -s http://localhost:11626/info | jq -r '.info.ledger.version'`" != "$PROTOCOL_VERSION" ]; do sleep 1; done
+    echo "upgrades: protocolversion done"
+  fi
+
+  # Upgrade local network's soroban config to match testnet
+  if [ $PROTOCOL_VERSION -ge 20 ]; then
+    echo "upgrades: soroban config"
+    echo $NETWORK_ROOT_SECRET_KEY | stellar-core get-settings-upgrade-txs "$NETWORK_ROOT_ACCOUNT_ID" 0 "$NETWORK_PASSPHRASE" --signtxs $COREHOME/etc/config-settings-entries.json 2>/dev/null | { \
+      read tx; \
+      read txid; \
+      echo "upgrades: soroban config: install contract: $txid .. $(curl -sG 'http://localhost:11626/tx' --data-urlencode "blob=$tx" | jq -r '.status')"; \
+      while [ "`curl -s http://localhost:11626/metrics | jq -r '.metrics."ledger.transaction.count".count'`" != "1" ]; do sleep 1; done
+      read tx; \
+      read txid; \
+      echo "upgrades: soroban config: deploy contract: $txid .. $(curl -sG 'http://localhost:11626/tx' --data-urlencode "blob=$tx" | jq -r '.status')"; \
+      while [ "`curl -s http://localhost:11626/metrics | jq -r '.metrics."ledger.transaction.count".count'`" != "2" ]; do sleep 1; done
+      read tx; \
+      read txid; \
+      echo "upgrades: soroban config: upload config: $txid .. $(curl -sG 'http://localhost:11626/tx' --data-urlencode "blob=$tx" | jq -r '.status')"; \
+      while [ "`curl -s http://localhost:11626/metrics | jq -r '.metrics."ledger.transaction.count".count'`" != "3" ]; do sleep 1; done
+      read key; \
+      echo "upgrades: soroban config: set config with key: $key";
+      curl -sG 'http://localhost:11626/upgrades?mode=set&upgradetime=1970-01-01T00:00:00Z' --data-urlencode "configupgradesetkey=$key"; \
+    }
+    echo "upgrades: soroban config done"
+  fi
+
+  # Delay friendbot starting till now because until network upgrades it won't be
```

Comment:
I think that the comment is wrong.. it should be ,
"Now that the network is ready, we can start off friendbot which would create accounts"

Reply:
Fixed in https://github.com/stellar/quickstart/pull/513/commits/3907ad87beea472199b1c04411e83a8dc34dcd00